__tests__/Guides.unit.test.ts

```diff
@@ -1,10 +1,560 @@
+    it('should create a guide from ruler, when drag is started from ruler', () => {
+      const element = document.createElement('canvas');
+      guidesInstance.ruler.canvasElement = element;
+      jest.spyOn(guidesInstance as any, 'createFromRuler').mockImplementation();
+      jest.spyOn(guidesInstance as any, 'offsetPosition').mockImplementation();
+      jest.spyOn(guidesInstance as any, 'matrix', 'get').mockImplementation();
+      const event = {
+        inputEvent: {
+          target: element,
+        },
+        datas: {
+          offsetPos: [1, 9],
+          matrix: 1,
+        },
+      };
+      (guidesInstance as any).dragStart(event as any);
+      expect((guidesInstance as any).createFromRuler).toHaveBeenCalled();
+    });
+
+    it('should create a guide from ruler', () => {
+      const event = {
+        datas: {},
+      };
+      jest.spyOn(guidesInstance as any, 'isLockType').mockReturnValue(false);
+      expect((guidesInstance as any).createFromRuler(event)).toBeTruthy();
+    });
+
+    it("shouldn't create a guide from ruler", () => {
+      const event = {
+        stop() {},
+      };
+      jest.spyOn(guidesInstance as any, 'isLockType').mockReturnValue(true);
+      expect((guidesInstance as any).createFromRuler(event)).toBeUndefined();
+    });
+
+    it('should change existing guide if this guide starts to drag', () => {
+      const element = document.createElement('div');
+      element.classList.add(GUIDE);
+      jest.spyOn(guidesInstance as any, 'startChangeExistGuide').mockImplementation();
+      jest.spyOn(guidesInstance as any, 'offsetPosition').mockImplementation();
+      jest.spyOn(guidesInstance as any, 'matrix', 'get').mockImplementation();
+      const event = {
+        inputEvent: {
+          target: element,
+        },
+        datas: {
+          offsetPos: [1, 9],
+          matrix: 1,
+        },
+      };
+      (guidesInstance as any).dragStart(event as any);
+      expect((guidesInstance as any).startChangeExistGuide).toHaveBeenCalled();
+    });
+
+    it('should change existing guide', () => {
+      const element = document.createElement('vic');
+      const event = {
+        stop() {},
+        inputEvent: {
+          target: element,
+        },
+        datas: {},
+      };
+      jest.spyOn(guidesInstance as any, 'isLockType').mockReturnValue(false);
+      expect((guidesInstance as any).startChangeExistGuide(event)).toBeFalsy();
+    });
+
+    it("shouldn't change existing guide", () => {
+      const element = document.createElement('vic');
+      const event = {
+        stop() {},
+        inputEvent: {
+          target: element,
+        },
+      };
+      jest.spyOn(guidesInstance as any, 'isLockType').mockReturnValue(true);
+      expect((guidesInstance as any).startChangeExistGuide(event)).toBeUndefined();
     });
+
+    it('should call dragStar function when drag triggers event onDragStart', () => {
+      defaultProps.lockGuides = false;
```

Comment:
since default props is the object from which we linked from the class - it is working.
But to make sure it is working 100% instead of link we should check the class property.
I think react clone default properties to the props